examples/isometric/game.go

```diff
@@ -0,0 +1,223 @@
+func (g *Game) Update() error {
+	// Update target zoom level.
+	var scrollY float64
+	if inpututil.KeyPressDuration(ebiten.KeyC) > 0 || inpututil.KeyPressDuration(ebiten.KeyPageDown) > 0 {
+		scrollY = -0.15
+	} else if inpututil.KeyPressDuration(ebiten.KeyE) > 0 || inpututil.KeyPressDuration(ebiten.KeyPageUp) > 0 {
+		scrollY = .15
+	} else {
+		_, scrollY = ebiten.Wheel()
+		if scrollY < -1 {
+			scrollY = -1
+		} else if scrollY > 1 {
+			scrollY = 1
+		}
+	}
+	g.camScaleTo += scrollY * (g.camScaleTo / 7)
+
+	// Clamp target zoom level.
+	if g.camScaleTo < 0.01 {
+		g.camScaleTo = 0.01
+	} else if g.camScaleTo > 100 {
+		g.camScaleTo = 100
+	}
+
+	// Smooth zoom transition.
+	div := 10.0
+	if g.camScaleTo > g.camScale {
+		g.camScale += (g.camScaleTo - g.camScale) / div
+	} else if g.camScaleTo < g.camScale {
+		g.camScale -= (g.camScale - g.camScaleTo) / div
+	}
+
+	pan := 4.0 / g.camScale
+	// Pan camera via keyboard.
+	if inpututil.KeyPressDuration(ebiten.KeyLeft) > 0 || inpututil.KeyPressDuration(ebiten.KeyA) > 0 {
+		g.camX -= pan
+	}
+	if inpututil.KeyPressDuration(ebiten.KeyRight) > 0 || inpututil.KeyPressDuration(ebiten.KeyD) > 0 {
+		g.camX += pan
+	}
+	if inpututil.KeyPressDuration(ebiten.KeyDown) > 0 || inpututil.KeyPressDuration(ebiten.KeyS) > 0 {
+		g.camY -= pan
+	}
+	if inpututil.KeyPressDuration(ebiten.KeyUp) > 0 || inpututil.KeyPressDuration(ebiten.KeyW) > 0 {
+		g.camY += pan
+	}
+	// Pan camera via mouse.
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButtonRight) {
+		if g.mousePanX == 0 && g.mousePanY == 0 {
+			g.mousePanX, g.mousePanY = ebiten.CursorPosition()
+		} else {
+			x, y := ebiten.CursorPosition()
+			dx, dy := float64(g.mousePanX-x)*(pan/100), float64(g.mousePanY-y)*(pan/100)
+			g.camX, g.camY = g.camX-dx, g.camY+dy
+		}
+	} else if g.mousePanX != 0 || g.mousePanY != 0 {
+		g.mousePanX, g.mousePanY = 0, 0
+	}
+
+	// Clamp camera position.
+	levelWidth := len(g.currentLevel.tiles)
+	levelHeight := len(g.currentLevel.tiles[0])
+	worldWidth := float64(levelWidth * g.currentLevel.tileSize / 2)
+	worldHeight := float64(levelHeight * g.currentLevel.tileSize / 2)
+	if g.camX < worldWidth*-1 {
+		g.camX = worldWidth * -1
+	} else if g.camX > worldWidth {
+		g.camX = worldWidth
+	}
+	if g.camY < worldHeight*-1 {
+		g.camY = worldHeight * -1
+	} else if g.camY > 0 {
+		g.camY = 0
+	}
+
+	// Randomize level.
+	if inpututil.IsKeyJustPressed(ebiten.KeyR) {
+		g.currentLevel = NewLevel()
+	}
+
+	return nil
+}
+
+// Draw draws the Game on the screen.
+func (g *Game) Draw(screen *ebiten.Image) {
+	// Render level.
+	g.renderLevel(screen)
+
+	// Print game info.
+	debugBox := image.NewRGBA(image.Rect(0, 0, g.w, 200))
+	debugImg := ebiten.NewImageFromImage(debugBox)
+	ebitenutil.DebugPrint(debugImg, fmt.Sprintf("KEYS WASD EC R\nFPS  %0.0f %c\nTPS  %0.0f\nSCA  %0.3f\nPOS  %0.1f,%0.1f", ebiten.CurrentFPS(), spinner[g.spinnerIndex], ebiten.CurrentTPS(), g.camScale, g.camX, g.camY))
+	g.op.GeoM.Reset()
```

Comment:
Do we need to save `op`? What about creating `DrawImageOptions` every time?

Reply:
Wouldn't that introduce an allocation each `Draw`?

Reply:
Creating a DrawImageOption every frame should not be so problematic (from my experience).